src/net/rpc.rs

```diff
@@ -1512,37 +1511,97 @@ impl ConversationHttp {
-                    warn!("Failed to load Stacks chain tip");
-                    let response = HttpResponseType::ServerError(
-                        response_metadata,
-                        format!("Failed to load Stacks chain tip"),
-                    );
-                    response.send(http, fd).and_then(|_| Ok(None))
+        if use_latest_tip {
+            let unconfirmed_chain_tip_opt = match &mut chainstate.unconfirmed_state {
+                Some(unconfirmed_state) => {
+                    if unconfirmed_state.is_readable() {
+                        // Check if underlying MARF trie exists before returning unconfirmed chain tip
+                        let trie_exists = match unconfirmed_state
+                            .clarity_inst
+                            .trie_exists_for_block(&unconfirmed_state.unconfirmed_chain_tip)
+                        {
+                            Ok(res) => res,
+                            Err(e) => {
+                                let response_metadata = HttpResponseMetadata::from(req);
+                                warn!("Failed to load Stacks chain tip; error checking underlying trie");
+                                let response = HttpResponseType::ServerError(
+                                    response_metadata,
+                                    format!("Failed to load Stacks chain tip: {:?}", e),
+                                );
+                                return response.send(http, fd).and_then(|_| Ok(None));
+                            }
+                        };
+
+                        if trie_exists {
+                            Some(unconfirmed_state.unconfirmed_chain_tip)
+                        } else {
+                            None
+                        }
+                    } else {
+                        None
+                    }
                 }
-            },
+                None => None,
+            };
+
+            if let Some(unconfirmed_chain_tip) = unconfirmed_chain_tip_opt {
+                Ok(Some(unconfirmed_chain_tip))
+            } else {
+                match chainstate.get_stacks_chain_tip(sortdb)? {
+                    Some(tip) => Ok(Some(StacksBlockHeader::make_index_block_hash(
+                        &tip.consensus_hash,
+                        &tip.anchored_block_hash,
+                    ))),
+                    None => {
+                        let response_metadata = HttpResponseMetadata::from(req);
+                        warn!("Failed to load Stacks chain tip");
+                        let response = HttpResponseType::ServerError(
+                            response_metadata,
+                            format!("Failed to load Stacks chain tip"),
+                        );
+                        response.send(http, fd).and_then(|_| Ok(None))
+                    }
+                }
+            }
+        } else {
```

Comment:
This API feels a little weird.  According to this code path, if I requested the query string `?tip=xxxxxxx&use_latest_tip=true`, not only would the request potentially succeed, but also the `tip=` argument would be ignored in favor of `use_latest_tip`.  Is this what we want?  Would it make more sense to require the caller to use *either* `tip=` *or* `use_latest_tip` but *not* both?

Reply:
It makes me wonder if instead of having a separate `use_latest_tip` query argument, we could instead overload `tip=` so that `tip=latest` becomes a valid argument.  Then, this ambiguity goes away.

Reply:
I think I like adding the option for `tip=latest`. I'll refactor this PR to do that.